### ISSUE TITLE
Allow changing envelope enryption for an EKS cluster

### ISSUE THREAD

**Issue by sagikazarmark** [banzaicloud/pipeline#3089]
**Is your feature request related to a problem? Please describe.**
As a user concerned about security, I want to change envelope encryption settings, particularly the KMS key.

**Describe the solution you'd like to see**
Add this feature to cluster update.

**Additional context**

- This is blocked by #3088
- It's not a priority at the moment
- Consider implementing this in the new control plane upgrade process
